plugins/jobs/src/js/JobDelete.js

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import PropTypes from "prop-types";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+
+import gql from "graphql-tag";
+
+import JobDeleteModal from "./components/JobDeleteModal";
+import defaultSchema from "./data/JobModel";
+
+const deleteJobMutation = gql`
+  mutation {
+    deleteJob(id: $jobId, stopCurrentJobRuns: $stopCurrentJobRuns) {
+      jobId
+    }
+  }
+`;
+
+function executeDeleteMutation({
+  jobId,
+  stopCurrentJobRuns,
+  onSuccess,
+  errorMessage
+}) {
+  return graphqlObservable(deleteJobMutation, defaultSchema, {
+    jobId,
+    stopCurrentJobRuns
+  })
+    .mapTo({ done: true, stopCurrentJobRuns, errorMessage })
+    .do(_ => onSuccess())
+    .startWith({ done: false, stopCurrentJobRuns, errorMessage });
+}
+
+function deleteEventHandler() {
+  const deleteSubject$ = new Subject();
+  const delete$ = deleteSubject$
+    .switchMap(executeDeleteMutation)
+    .catch(error => {
+      return delete$.startWith({
+        errorMessage: error && error.response && error.response.message,
+        done: true
+      });
+    });
+
+  return {
+    delete$,
+    deleteHandler: (jobId, stopCurrentJobRuns, onSuccess, errorMessage) => {
+      deleteSubject$.next({
+        jobId,
+        stopCurrentJobRuns,
+        onSuccess,
+        errorMessage
+      });
+    }
+  };
+}
+
+function isTaskCurrentRunning(errorMessage) {
```

Comment:
I hate to get back to this but Task preceding Current still doesn't make any sense to me. Am I missing something?

Reply:
I think what is wrong here is that it should be written "currently". Does that work for you? `isTaskCurrentlyRunning`

Reply:
that would make total sense 🙇